libimage/push.go

```diff
@@ -85,5 +87,21 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 
 	defer c.close()
 
-	return c.copy(ctx, srcRef, destRef)
+	manifestBytes, err := c.copy(ctx, srcRef, destRef)
+	if err != nil {
+		return nil, err
+	}
+	manifestDigest, err := manifest.Digest(manifestBytes)
+	if err != nil {
+		return nil, err
+	}
+	// Tag for pushed remote digest i.e repo@remote-digest
+	remoteDigest := strings.TrimPrefix(manifestDigest.String(), "sha256:")
+	pushedTag := destRef.DockerReference().Name() + ":" + remoteDigest
+	logrus.Debugf("Successfully pushed image and now creating local tag %q", pushedTag)
+	err = image.Tag(pushedTag)
```

Comment:
I don't think that _tagging_ is the right thing here.  Shouldn't the digest be part of `storage.Image.Digests`?

Reply:
Looking at all the digests is part of image lookup (see https://github.com/containers/common/blob/6d22aa8cbd3f695c0228aa62a0d136306ac833ee/libimage/runtime.go#L403).  `buildah inspect repo:digest` will hence work.

Reply:
ah i see but looks like we would need new API in c/storage to update image digests for this so we don't end up creating image again.

Reply:
Yes, we need such an API.  In theory, tag and untag would need such an API as well.  I opened an issue against c/storage a while ago for that: https://github.com/containers/storage/issues/999

Reply:
@vrothberg for Tag and Untag could we still use `AddNames`, `RemoveNames` ? but for `digests` i guess we have nothing, I think we will have to close this first https://github.com/containers/storage/issues/999 I'll take a stab at it.

Reply:
Yes, this should use ordinary digest references. It should have substantially the same outcome as `podman pull repo@sha256:$digest`, which also adds a digest name to the image.

Inventing a new tag-with-digest convention is both unnecessary and counterproductive to https://github.com/containers/storage/issues/999 ’s goal of tracking digests correctly.

Reply:
c/storage’s `AddNames` is already used to add digest references (compare the `pull` case described above), no new API is necessary for that. https://github.com/containers/storage/issues/999 primarily talks about c/storage maintaining a separate list of tagged/named references, and potentially making them available separately and the like, but nothing about that requires a change of the `AddNames` API AFAICS.

Reply:
… actually, `AddNames` is correct for adding digested references but insufficient, we do need to actually record the pushed manifest. More in https://github.com/containers/storage/issues/999 .